pkg/server/server.go

```diff
@@ -673,6 +673,27 @@ func (s *Server) indexHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// This Content Security Policy (CSP) applies to Console web application resources.
+	// Console CSP is deployed in report-only mode via "Content-Security-Policy-Report-Only" header.
+	// See https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP for details on CSP specification.
+	cspSources := "'self'"
+	if s.K8sMode == "off-cluster" {
+		// Console local development involves a webpack server running on port 8080
+		cspSources = cspSources + " http://localhost:8080 ws://localhost:8080"
+	}
+	cspDirectives := []string{
+		fmt.Sprintf("default-src %s", cspSources),
+		fmt.Sprintf("base-uri %s", cspSources),
+		fmt.Sprintf("img-src %s data:", cspSources),
+		fmt.Sprintf("font-src %s data:", cspSources),
```

Comment:
sorry, what does an empty `data:` schema mean here?

Reply:
@yapei This allows the web application to use [`data:` URLs](https://developer.mozilla.org/en-US/docs/Web/URI/Schemes/data) to embed certain content into the HTML markup.

In this case, we allow using `data:` URLs for [`img-src`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/img-src) (images and favicons) and [`font-src`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/font-src) (CSS fonts).

Using `data:` URLs is commonly used for web asset optimization. When building Console web application, the module bundler (webpack) may generate these `data:` URLs for any small-sized images and fonts.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/Sources#scheme-source

Reply:
yes, this should be a valid value 👍